addon/components/ember-thead/component.js

```diff
@@ -36,77 +31,81 @@ import layout from './template';
   @yield {object} h - the API object yielded by the table header
   @yield {Component} h.row - The table row component
 */
-@tagName('thead')
-export default class EmberTHead extends Component {
+
+export default Component.extend({
+  layout,
+  tagName: 'thead',
+
   /**
     The API object passed in by the table
+
+    @argument
+    @required
+    @type object
   */
-  @argument
-  @required
-  @type('object')
-  api;
+  api: null,
 
-  @or('api.api', 'api')
-  unwrappedApi;
+  unwrappedApi: or('api.api', 'api'),
 
   /**
     The column definitions for the table
+
+    @argument
+    @required
+    @type(Array)
   */
-  @argument
-  @required
-  @type(Array)
-  columns;
+  columns: defaultTo(() => []),
 
   /**
     An ordered array of the sorts applied to the table
+    @argument({ defaultIfUndefined: true })
+    @type(Array)
   */
-  @argument({ defaultIfUndefined: true })
-  @type(Array)
-  sorts = [];
+  sorts: defaultTo(() => []),
 
   /**
     An optional sort
+    @argument({ defaultIfUndefined: true })
+    @type(optional(Function))
   */
-  @argument({ defaultIfUndefined: true })
-  @type(optional(Function))
-  sortFunction = sortMultiple;
+  sortFunction: sortMultiple,
 
   /**
     An ordered array of the sorts applied to the table
+    @argument({ defaultIfUndefined: true })
+    @type(optional(Function))
   */
-  @argument({ defaultIfUndefined: true })
-  @type(optional(Function))
-  compareFunction = compareValues;
+  compareFunction: compareValues,
```

Comment:
I tried adding `defaultTo` and had lots of failures.

Reply:
ah, it's trying to run it because it's a function, it treats it like an initializer. So, kind of counterintuitive, but you'd have to do:

```js
defaultTo(() => compareValues),
```

If no tests are failing maybe it's not an issue, but it definitely would be different than the way other args are defaulted

Reply:
I tried that and it still throws the errors. I'm curious, why do we need `defaultTo`? Wouldn't what we set be the default, and then anything passed to the component override it?

Reply:
It covers two cases:

1. When you pass `undefined` as the argument. This typically happens as a result of _prop drilling_ when you're trying to wrap a component and are forwarding all of the arguments from the parent. The parent doesn't have defaults defined, so it defaults to `undefined`, which then overwrites your default.

2. When you _update_ the argument to be `undefined` for some reason. This is much less common, but does happen. If you update the argument in the external space to be undefined, maybe because you weren't thinking about what it would do to your children, that's typically not the behavior you want. The CP will provide the default instead.

We use the prop drilling in tests pretty frequently, because we have a default table template that has every prop value passed into it. I'll see if I can get it to work.